Code/Framework/AzToolsFramework/AzToolsFramework/Logger/TraceLogger.h

```diff
@@ -55,7 +55,8 @@ namespace AzToolsFramework
             AZStd::string window;
             AZStd::string message;
         };
-        AZStd::vector<LogMessage> m_startupLogSink;
+
+        AZStd::list<LogMessage> m_startupLogSink;
```

Comment:
maybe a `deque`  could serve here?

I'm not a fan of lists :P

Reply:
Not very common to see lists used in this codebase but it seemed like the right container for the job since I'm only pushing to the end and not doing any random access.

Reply:
This was a good suggestion though, deque does a good job balancing dynamic allocation while still remaining somewhat cache efficient. My guess is, if your changes were performance inspired, deque would actually perform better than list here. Did you do any profiling or benchmarking with these changes?